packages/cli/src/commands/transfer/celo.ts

```diff
@@ -22,42 +23,80 @@ export default class TransferCelo extends BaseCommand {
   ]
 
   async run() {
-    const kit = await this.getKit()
+    const client = await this.getPublicClient()
+    const wallet = await this.getWalletClient()
     const res = await this.parse(TransferCelo)
 
+    if (!wallet) {
+      throw new Error('TODO: only AKV doesnt return a wallet')
+    }
+
     const from = res.flags.from
     const to = res.flags.to
-    const value = new BigNumber(res.flags.value)
+    const value = res.flags.value
+    const feeCurrency = res.flags.gasCurrency
+
+    const params = feeCurrency ? { feeCurrency } : {}
 
-    kit.defaultAccount = from
-    const celoToken = await kit.contracts.getGoldToken()
+    const goldTokenContract = await getGoldTokenContract(client as PublicClient)
+    const transferWithCommentContractData = {
+      address: goldTokenContract.address,
+      abi: goldTokenContract.abi,
+      account: from,
+      functionName: 'transferWithComment',
+      args: [to, value, res.flags.comment!],
+      ...params,
+    } as const
+
+    const transferParams = { to, value: value, ...params } as const
 
     await newCheckBuilder(this)
       .isNotSanctioned(from)
       .isNotSanctioned(to)
+      .isValidWalletSigner(from)
       .hasEnoughCelo(from, value)
-      .runChecks()
+      .addCheck(
+        `Account can afford to transfer CELO with gas paid in ${feeCurrency || 'CELO'}`,
+        async () => {
+          const [gas, gasPrice, balanceOfTokenForGas] = await Promise.all([
+            res.flags.comment
+              ? client.estimateContractGas(transferWithCommentContractData)
+              : client.estimateGas(transferParams),
+            client.getGasPrice(),
```

Comment:
isnt this missing the price of gas in the gas token? ie feeCurrency needs to be provided as a param to getGasPrice (or a different function needs to be called)